common/net/bolt1.py

```diff
@@ -0,0 +1,268 @@
+    """
+
+    def __init__(self, mtype, payload, extension=None):
+        self.type = mtype
+        self.payload = payload
+        self.extension = extension
+
+    @classmethod
+    def from_bytes(cls, message):
+        """
+        Builds a message from its byte representation.
+
+        Args:
+            message (:obj:`bytes`): the byte-encoded message.
+
+        Returns:
+            The Message children class depending on the received message type. Check ``message_types`` for more info.
+
+        Raises:
+            :obj:`TypeError`: If the given message is not in bytes.
+            :obj:`ValueError`: If the message can not be parsed.
+        """
+
+        if not isinstance(message, bytes):
+            raise TypeError(f"message be must a bytearray")
+        if len(message) < 2:
+            raise ValueError(f"message be must at least 2-byte long")
+
+        if message[:2] == message_types["init"]:
+            return InitMessage.from_bytes(message)
+        elif message[:2] == message_types["error"]:
+            return ErrorMessage.from_bytes(message)
+        elif message[:2] == message_types["ping"]:
+            return PingMessage.from_bytes(message)
+        elif message[:2] == message_types["pong"]:
+            return PongMessage.from_bytes(message)
+
+    def serialize(self):
+        """Serialises the message."""
+        if not self.extension:
+            return self.type + self.payload
+        else:
+            tlvs = b"".join([tlv.serialize() for tlv in self.extension()])
+            return self.type + self.payload + tlvs
+
+
+class InitMessage(Message):
+    """
+    First message exchange by the nodes, it reveals the features supported by each end.
+
+    Args:
+        global_features (:obj:`FeatureVector <teos.net.bolt9.FeatureVector>`): the global features vector.
+        local_features (:obj:`FeatureVector <teos.net.bolt9.FeatureVector>`): the local features vector.
+        local_features (:obj:`list`): a list of genesis block hashes (optional).
+    """
+
+    def __init__(self, global_features, local_features, networks=None):
+        if not (isinstance(global_features, FeatureVector) and isinstance(local_features, FeatureVector)):
+            raise TypeError("global_features and local_features must be FeatureVector instances")
+        if networks:
+            if not isinstance(networks, NetworksTLV):
+                raise TypeError("networks must be of type NetworksTLV (if set)")
+
+        global_features = global_features.serialize()
+        local_features = local_features.serialize()
+        gflen = len(global_features).to_bytes(2, "big")
+        flen = len(local_features).to_bytes(2, "big")
+        payload = gflen + global_features + flen + local_features
+
+        # Add extensions if needed (this follows TLV format)
+        # FIXME: Only networks for now
+        if networks:
+            super().__init__(mtype=message_types["init"], payload=payload, extension=networks.serialize())
```

Comment:
Seems you don't distinguish between a single TLV and a TLV stream, that is ok for now, but once we get to streams with multiple TLVs that might put more work on the caller.